### ISSUE TITLE
validation and testing accuracy on a complex model

### ISSUE THREAD

**Issue by deepdata1010** [fchollet/keras#5570]
Hi, I create a complex model using the functional api Model by combining two sequential models. When the model is training using model.fit,
the validation accuracy  on the validation dataset is m, but when I use the model.evaluate to test the accuracy of the validation dataset, it gives me another value.  I am wondering which one is correct and how to test the data correctly for the complex model. Thanks.

**Comment by patyork** [comment]
You'll need to post an example that shows this discrepancy.

A quick test on completely random data, using a merged model as well:

Train time, final epoch:
```
Epoch 10/10
5000/5000 [==============================] - 0s - loss: 0.0841 - val_loss: 0.0853
```
Evaluate time:
```
0.0853 49295139312747
```

**Comment by deepdata1010** [comment]
Hi, 

Thanks. The main code is as follows. I tried to combine a siamese network and a classification network.


basenet11=create_base_network11(input_dim)
basenet12=create_base_network12(4096)
base_network1 = create_base_network(basenet11,basenet12,input_dim)

input_a = Input(shape=(input_dim,))
input_b = Input(shape=(input_dim,))

processed_a = basenet11(input_a)
processed_b = basenet11(input_b)

basenet22=create_base_network22(input_dim2)
pre = basenet22(processed_a)

model = Model(input=[input_a, input_b], output=[distance,pre])

sgd = SGD(lr=0.001, decay=1e-6, momentum=0.9, nesterov=True)

model.compile(optimizer=sgd,  loss=[  contrastive_loss,  'categorical_crossentropy'],
              loss_weights=[1., 1.], metrics=[ 'accuracy'])
 
model.fit([tr_pairs[:, 0], tr_pairs[:, 1]],[tr_y,tr_yall],
          validation_data=([tv_pairs[:, 0], tv_pairs[:, 1]],[ tv_y,tv_yall]),
          batch_size=128,
          nb_epoch=nb_epoch) 

 
score =model.evaluate([tv_pairs[:, 0], tv_pairs[:, 1]],[ tv_y,tv_yall])
print('Test score:', score )

The results are as follows:

Train on 8400 samples, validate on 112 samples
Epoch 1/1
8400/8400 [==============================] - 239s - loss: 9.5694 - lambda_1_loss: 8.9944 - sequential_4_loss: 0.5750 - lambda_1_acc: 0.4664 - sequential_4_acc: 0.8313 - val_loss: 2.6879 - val_lambda_1_loss: 0.0000e+00 - val_sequential_4_loss: 2.6879 - val_lambda_1_acc: 0.0000e+00 - val_sequential_4_acc: 0.2768
112/112 [==============================] - 3s     
('Test score:', [3.0492388180324008, 0.0, 3.0492388180324008, 0.0, 0.19642857142857142])

**Comment by deepdata1010** [comment]
Do you just use the model.evaluate ? Thanks.
